### PR TITLE
Add howto page describing how to configure network of hosted cluster

### PR DESCRIPTION
**Pull request by thallgren**
Adds a new howto page that describes the problem and explains how
to solve it along with a complete example. Also adds a pointer to this
document from the troubleshooting page.

### CODE REVIEW

File: docs/pre-release/doc-links.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@
        link: install/upgrade/
      - title: Install Traffic Manager with Helm
        link: install/helm/
+     - title: Telepresence Pro
+       link: install/telepresence-pro/
```
Comment (richarddli) [inline docs/pre-release/doc-links.yml]: I'm not clear from this PR why we need to add Telepresence Pro?

File: docs/pre-release/doc-links.yml [hunk 2]
```diff
@@ -32,6 +34,8 @@
        link: howtos/outbound
      - title: Send requests to an intercepted service
        link: howtos/request
+     - title: Use Telepresence with a cluster hosted in a VM on the workstation
```
Comment (richarddli) [inline docs/pre-release/doc-links.yml]: How about "Use Telepresence with a local Kubernetes cluster" (I know that is less technically accurate but I don't think most of our users understand how k3s works in a VM for example)

File: docs/pre-release/howtos/cluster-in-vm.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
+---
+description: "Use Telepresence to intercept services in a cluster running in a hosted virtual machine."
+---
+
+# Network considerations for locally hosted clusters
+
+## The problem
+Telepresence creates a Virtual Network Interface ([VIF](../tun-device)) that maps the clusters subnets to the host machine when it connects. This is not a problem under normal circumstances (when the cluster runs in the cloud), but network problems may arise when the cluster runs in a VM that is hosted on the same machine where Telepresence runs.
```
Comment (richarddli) [inline docs/pre-release/howtos/cluster-in-vm.md]: On the second sentence, I'd reword:

"If you're running Kubernetes locally (e.g., k3s, Minikube, Docker for Desktop), you may encounter network problems ..."

File: docs/pre-release/howtos/cluster-in-vm.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+Now, if a request arrives to Telepresence that is covered by a subnet mapped by the VIF, the request is routed to the cluster. If the cluster for some reason doesn't find a corresponding listener that can handle the request, it will eventually try the host network, and find the VIF. The VIF routes the request to the cluster and now the recursion is in motion. The final outcome of the request will likely be a timeout but since the recursion is very resource intensive (a large amount of very rapid connection requests), this will likely also affect other connections in a bad way. 
+
+## Solution
+
+### Detect and kill recursion?
+A recursion detection mechanism was introduced in Telepresence 2.4.11 that attempted to prevent the recursion by detecting successive attempts to connect to the same address before a previous attempt had succeeded. It worked well for simple use-cases but performed badly when a large number of connections were established as a consequence of opening a webpage. Since all attempts to the same address were serialized, some of them timed out. The detection mechanism was therefore removed in 2.5.3.
```
Comment (richarddli) [inline docs/pre-release/howtos/cluster-in-vm.md]: I'd delete this section, or move it to the bottom, since it doesn't help the reader get to the solution.

File: docs/pre-release/howtos/cluster-in-vm.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+### A better solution
+The network of a VM running on the host can often be configured in different ways. In the example above, the "host-only" network can be replaced by a "bridge" network. A bridge will attach directly to a network interface card on the host and bypass the host's network stack. In essence, it makes the VM connect directly to the same router as the host. The Telepresence VIF is not visible to that router but the router will allow its connected instances to communicate. This solves the problem, because even if a request intended for the cluster may be sent to the router, the router will never send it to the host.
+
```
Comment (richarddli) [inline docs/pre-release/howtos/cluster-in-vm.md:21]: Instead of "a better solution", how about something like:

### Create a bridge network

A bridge network is a Link Layer (L2) device that forwards traffic between network segments. By creating a bridge network, you can bypass the host's network stack which enable the Kubernetes cluster to connect directly to the same router as your host. 

To create a bridge network, you need to XXX ...

File: docs/pre-release/doc-links.yml (hunk 1 above)
Comment (thallgren) [reply docs/pre-release/doc-links.yml]: Oops. That change shouldn't be there.

**Review by richarddli — APPROVED** [review]
